packages/eslint-plugin/src/types/rules.ts

```diff
@@ -1,34 +1,8 @@
-import { TSESTree } from "@typescript-eslint/experimental-utils";
+import { TSESTree } from "@typescript-eslint/typescript-estree";
 import { AST, Linter, Rule, Scope, SourceCode } from "eslint";
-type CodePath = Rule.CodePath;
-type CodePathSegment = Rule.CodePathSegment;
 
 interface TSRuleListener {
-  onCodePathStart?(codePath: CodePath, node: TSESTree.Node): void;
-
-  onCodePathEnd?(codePath: CodePath, node: TSESTree.Node): void;
-
-  onCodePathSegmentStart?(segment: CodePathSegment, node: TSESTree.Node): void;
-
-  onCodePathSegmentEnd?(segment: CodePathSegment, node: TSESTree.Node): void;
-
-  onCodePathSegmentLoop?(
-    fromSegment: CodePathSegment,
-    toSegment: CodePathSegment,
-    node: TSESTree.Node,
-  ): void;
-
-  [key: string]:
-    | ((node: TSESTree.ImportDeclaration) => void)
-    | ((node: TSESTree.CallExpression) => void)
-    | ((codePath: CodePath, node: TSESTree.Node) => void)
-    | ((segment: CodePathSegment, node: TSESTree.Node) => void)
-    | ((
-        fromSegment: CodePathSegment,
-        toSegment: CodePathSegment,
-        node: TSESTree.Node,
-      ) => void)
-    | undefined;
+  [key: string]: (() => void) | ((node: TSESTree.Node) => void) | undefined;
```

Comment:
I'm not sure if I understand this type. The `undefined` is unnecessary, and the first case is telling us that sometimes the listener doesn't take an argument?